docs/proposals/user-impersonation-proposal.md

```diff
@@ -0,0 +1,39 @@
+# User Impersonation Proposal
+
+## Introduction
+We will need to add some security settings to help administrators lock down the broker. For this, we are going to check for privilege escalation and will move the location of where the APB pod will run.
+
+## Problem Description
+The problem is that currently privilege escalation is a concern for users who have access to the broker. We want to give the cluster admin who is setting up the service broker the ability to not allow users to have privilege escalation. 
+
+## Implementation Details
+We will need to do 5 things to satisfy the requirements.
+1. Check if the user has the permissions to cover the cluster role to be used.
+2. If the cluster admin while setting up the broker has set a config value to auto escalate we will immediately continue and not check the user's permissions.
```

Comment:
I would combine 1+2, and make it something like

Check if the user has the permissions to cover the cluster role to be used. If the auto-escalate option is enabled in the broker config, we will immediately continue and not check the user's permissions.